scripts/asm-installer/install_asm

```diff
@@ -728,6 +733,47 @@ validate_istio_version() {
   fi
 }
 
+validate_asm_version() {
+  info "Checking existing ASM version(s)..."
+  local OUTPUT
+  OUTPUT="$(retry 3 ./istio-${RELEASE}/bin/istioctl version -o json)"
+  local FOUND_INVALID_VERSION
+  FOUND_INVALID_VERSION="false"
+  for version in $(echo "${OUTPUT}" | jq -r '.meshVersion[].Info.version'); do
+    if ! [[ "$version" =~ "asm" ]]; then
+      fatal "Cannot upgrade from version $version. Only upgrade from ASM distribution is supported."
+    fi
+
+    # if asm version found, pattern: 1.6.11-asm.1-586f900508ad482ed32b830dd15f6c54b32b93ed
+    local VERSION_MAJOR VERSION_MINOR VERSION_POINT VERSION_REV
+    IFS="." read VERSION_MAJOR VERSION_MINOR VERSION_POINT VERSION_REV <<< "$version"
+    VERSION_POINT="$(echo "$VERSION_POINT" | sed 's/-.*//')"
+    VERSION_REV="$(echo "$VERSION_REV" | sed 's/-.*//')"
+    if [[ "$VERSION_MAJOR" -ne 1 || "$VERSION_MINOR" -lt 6 || "$VERSION_MINOR" -ge "$MINOR" || \
+      ( "$VERSION_MINOR" -eq "$MINOR" && \
+        ( "$VERSION_POINT" -ge "$POINT" || \
+          ("$VERSION_POINT" -eq "$POINT" && "$VERSION_REV" -ge "$REV"))) ]]; then
+      info "  $version (not suitable for migration)"
+      FOUND_INVALID_VERSION="true"
+    else
+      info "  $version (suitable for migration)"
+    fi
+  done
+  if [[ "$FOUND_INVALID_VERSION" = "true" ]]; then
+    fatal "Upgrade requires all existing control planes to be between versions 1.6.0-asm.0 (inclusive) and ${RELEASE} (exclusive)."
+  fi
+}
+
+validate_ca_consistency() {
+  info "Checking existing CA..."
+  local INSTALLED_CA
+  INSTALLED_CA="$(kubectl -n istio-system get pod -l istio=ingressgateway -o jsonpath='{.items[].spec.containers[].env[?(@.name=="CA_ADDR")].value}')"
+  if [[ ("${CA}" = "citadel" && ! "${INSTALLED_CA}" =~ istiod.*istio\-system\.svc) || \
+        ("${CA}" = "mesh_ca" && ! "${INSTALLED_CA}" =~ meshca\.googleapis\.com) ]]; then
+    fatal "CA cannot be switched while performing upgrade. Please use the same CA."
```

Comment:
We know the CA, instead of "same CA" we should use the value.

Reply:
I like the suggestion, thanks!